Document User Attributes Display Name and Name (v11.8) (#9025)
mattermost/docs: source/administration-guide/manage/admin/user-attributes.rst

Reviewer: We should add here `Display Name` as the first item they would need to specify and then for the `Name` we should indicate that it auto fills from the display name and that is CEL Safe. we should include what those restrictions are somehwehere: 

- Must start with a letter (A–Z, a–z) or underscore (_)
- Remaining characters may only be letters, digits (0–9), or underscores
- No spaces, hyphens, dots, or other punctuation
- It should not be reserved CEL keywords: “true”, “false”, “null”, “in”, “as”, “break”, “const”, “continue”, “else”, “for”, “function”, “if”, “import”, “let”, “loop”, “package”, “namespace”, “return”, “var”, “void”, “while”

```suggestion
    
    -- **Display Name**: Enter a display name for the attribute that will be shown...  
    - **Attribute name**: The attribute name is automatically generated as a snake case name from the Display Name. This can be modified to any unique name CEL-safe, that should not include 
```

From Mattermost v11.8, user attributes include both a **Display Name** and a **Name**. The Display Name is the label shown to users and admins in Mattermost. The Name is the internal attribute identifier used in API references and :doc:`attribute-based access control (ABAC) </administration-guide/manage/admin/attribute-based-access-control>` policy expressions. When you create a new user attribute, Mattermost can generate the Name from the Display Name, formatted as a CEL-safe identifier. Saved ABAC policies continue to reference the internal name using ``user.attributes.<name>``.

.. note::

  When you disable the **Editable by Users** option for an attribute, only admins can set its value using :ref:`mmctl cpa <administration-guide/manage/mmctl-command-line-tool:mmctl cpa>` commands.

1. In the System Console, go to **Site Configuration > System Attributes > User Attributes** and select **Add Attribute**.
2. Enter the following details:

>     

Reviewer: Is this extra line intended?

Author: Nope! Thank you again.

>     - **Display Name**: Enter a display name for the attribute that will be shown.  

Reviewer: ```suggestion
    - **Display Name**: Enter a display name for the attribute that will be shown in any user facing UI.  
```

>     -- **Display Name**: Enter a display name for the attribute that will be shown...  

Reviewer: My suggestion was not complete. It was an indication of what we should include

Author: Thanks @marianunez! I updated the Attribute name description to include the CEL-safe naming requirements, including the allowed character rules and reserved CEL words list. I think that should cover it?

>     - **Attribute name**: Enter a unique name for the attribute. Attribute names can be up to 40 characters long. The attribute name is used as the Common Expression Language (CEL) identifier in access control policies. It must start with a letter or underscore, and can contain only letters, numbers, and underscores. Reserved CEL words aren't allowed, including ``true``, ``false``, ``null``, ``in``, ``as``, ``break``, ``const``, ``continue``, ``else``, ``for``, ``function``, ``if``, ``import``, ``let``, ``loop``, ``package``, ``namespace``, ``return``, ``var``, ``void``, and ``while``. 

Reviewer: ```suggestion
    - **Attribute name**: Attribute name will be automatically generated from the provided Display Name. It can be overriden and with a preferred unique name for the attribute. Attribute names can be up to 40 characters long. The attribute name is used as the Common Expression Language (CEL) identifier in access control policies. It must start with a letter or underscore, and can contain only letters, numbers, and underscores. Reserved CEL words aren't allowed, including ``true``, ``false``, ``null``, ``in``, ``as``, ``break``, ``const``, ``continue``, ``else``, ``for``, ``function``, ``if``, ``import``, ``let``, ``loop``, ``package``, ``namespace``, ``return``, ``var``, ``void``, and ``while``. 
```
